src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -31,16 +33,38 @@ private List<object> Buffer
                     _buffer = new List<object>();
                 }
 
+                if (_isSingleContentSet)
+                {
+                    Debug.Assert(_buffer.Count == 0);
+
+                    _buffer.Add(_singleContent);
+                    _isSingleContentSet = false;
+                }
```

Comment:
No need to check `_isSingleContentSet` except inside the `_buffer == null` block. IOW don't un-nest the code just to be able to perform the assertion.